### PR TITLE
Implemented look wise boundary recomputation in PC analysis

### LINKED ISSUE #127: Implement adaptive boundary recomputation based on actual information fractions in non-interactive PC analysis API
**Scope**
- Implement this only for the non-interactive PC API.
- Exclude legacy interactive API changes (planned for deprecation).

### Problem
Current PC analysis uses boundaries computed once from planned IF and then applies them statically. This can misrepresent error spending when actual look timing deviates from plan.

### Goal
At each non-interactive PC look, recompute boundaries using actual IF observed so far plus planned IF for remaining looks.

### Key decisions
1. Actual per-look IF is mandatory in analysis calls.
2. Default IF tolerance is 0.05 and is configurable.
3. Finality is IF-driven, not planned-look-count-driven:
   - `is_final_look <- current_if >= (1.0 - tolerance)`
4. Interim IF rule is strict:
   - Interim look iff `current_if < (1.0 - tolerance)`
   - Interim IF must satisfy `0 < current_if < (1.0 - tolerance)`
5. Final IF rule:
   - Final IF must satisfy `current_if <= (1.0 + tolerance)`
   - So valid final range is `[(1.0 - tolerance), (1.0 + tolerance)]`
6. IF monotonicity applies at all looks:
   - `current_if` must be strictly greater than previous IF.

### Implementation details
- Add `info_frac_tolerance` to setup API (default 0.05), store in design params/state.
- Add mandatory `info_frac_current` to non-interactive look analysis API.
- On each look:
  1. Validate IF (strict interim rule above + final upper bound + monotonicity).
  2. Reconstruct IF vector as:
     - actual IFs for completed looks,
     - planned IFs for not-yet-completed looks.
  3. Recompute design via `rpact::getDesignGroupSequential(...)`.
  4. Update current look p-value cutoff and inverse-normal weights from recomputed design.
  5. Persist actual IF in state history.

### Edge cases to support
- Off-schedule looks (actual IF differs from planned IF within tolerance).
- Unplanned timing patterns.
- Over-recruitment at final look (`IF > 1.0` but `<= 1.0 + tolerance`).

### Out of scope
- Interactive/legacy PC analysis API updates.
- PE-PC wrapper changes (to be handled in follow-up issue).

### Acceptance criteria
- Non-interactive setup accepts/stores `info_frac_tolerance`.
- Non-interactive look analysis requires `info_frac_current`.
- Interim validation is strict: `0 < IF < 1.0 - tolerance`.
- Finality begins at `IF >= 1.0 - tolerance`.
- Final IF upper bound enforced: `IF <= 1.0 + tolerance`.
- IF strict monotonicity enforced across looks.
- Per-look rpact boundary recomputation implemented and used.
- State tracks actual IF history for future recomputation.
- Tests cover interim/final boundary classification and IF edge conditions.

### PR DESCRIPTION
**Pull request by DeshmukhOnLine**
Resolves #127

### CODE REVIEW

File: R/PcAnalysisApi.R [hunk 1]
```diff
@@ -4,6 +4,97 @@
...
+    }
+
+    return(rpact::getDesignGroupSequential(
+      kMax = k_max,
+      alpha = design_params$alpha,
+      informationRates = information_rates,
+      typeOfDesign = type_of_design,
+      userAlphaSpending = spending
+    ))
+  }
```
[the remark this answers is not in the document]
Comment (DeshmukhOnLine) [reply R/PcAnalysisApi.R:61]: I am not sure about your suggestion. I will handle this case later. I have reported bug #130 for this.

File: R/PeAnalysisApi.R [hunk 2]
```diff
@@ -370,11 +377,18 @@ AnalyzeLook_PE_PC <- function(
 
   nPreviousLooks <- state$completed_looks
 
+  if (is.null(info_frac_cur)) {
+    nNextLook <- state$completed_looks + 1L
+    nPlannedLooks <- length(state$design_params$info_frac)
+    info_frac_cur <- state$design_params$info_frac[min(nNextLook, nPlannedLooks)]
+  }
```
[the remark this answers is not in the document]
Comment (DeshmukhOnLine) [reply R/PeAnalysisApi.R:384]: This changes in PE-PC is temporary since we needed its tests to pass. We will correct this later as a part of PE-PC analysis changes. In this PR, we are focusing on PC analysis.
